app/src/main/java/com/arctouch/codechallenge/ui/home/HomeAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package com.arctouch.codechallenge.ui.home
+
+import android.databinding.DataBindingUtil
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.databinding.MovieItemBinding
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.ui.details.MovieDetailsActivity
+import com.arctouch.codechallenge.ui.home.movie.MovieViewModel
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import kotlinx.android.synthetic.main.movie_item.view.*
+
+class HomeAdapter : RecyclerView.Adapter<HomeAdapter.ViewHolder>() {
+
+    private lateinit var movieList: MutableList<Movie>
+
+    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): ViewHolder {
+        val binding: MovieItemBinding =
+                DataBindingUtil.inflate(LayoutInflater.from(p0.context), R.layout.movie_item, p0, false)
+
+        return ViewHolder(binding)
+    }
+
+    override fun getItemCount() = if (::movieList.isInitialized) movieList.size else 0
+
+    fun updatePartnerList(partnerList: List<Movie>) {
+        if (::movieList.isInitialized) movieList.addAll(partnerList) else movieList = partnerList.toMutableList()
```

Comment:
Why did you choose to make `movieList` a `lateinit var` if there could be issues with its initialisation? Wouldn't a (maybe nullable) `var` be better?

Reply:
Because the MutableList can only be initialized by knowing the size you will have on it and the data. And we don`t have it until the request is finished. So basically I`m starting the adapter with no data and updating it if there`s some result from the request. And I always check if the movieList is initialized, otherwise the adapter return no items.